### PR TITLE
Support RBAC

### PR DESCRIPTION
**Pull request by monstermunchkin**

### CODE REVIEW

File: lxd/daemon.go [hunk 1]
```diff
@@ -755,15 +764,48 @@ func (d *Daemon) init() error {
...
 	if err != nil {
 		return err
 	}
 
-	err = d.setupExternalAuthentication(candidEndpoint, candidEndpointKey, candidExpiry, candidDomains)
-	if err != nil {
-		return err
+	// Ensure that not both Candid and RBAC are configured at the same time
+	if candidEndpoint != "" && rbacAPIURL != "" {
+		logger.Warn("Disabling Candid and RBAC since they cannot be configured simultaneously")
```
Comment (stgraber) [inline lxd/daemon.go]: This must be caught earlier

File: lxd/daemon.go [hunk 2]
```diff
@@ -755,15 +764,48 @@ func (d *Daemon) init() error {
...
+	} else {
+		if rbacAPIURL != "" {
+			err = d.setupExternalAuthentication(rbacAuthURL, "", rbacExpiry, "")
```
Comment (stgraber) [inline lxd/daemon.go]: setupRBAC should do that

File: lxd/daemon.go [hunk 3]
```diff
@@ -755,15 +764,48 @@ func (d *Daemon) init() error {
...
+			go func() {
+				for {
+					err = d.setupRBACServer(rbacAPIURL, rbacAPIKey, rbacAPIUsername, rbacAgentPrivateKey, rbacAgentPublicKey, rbacAuthURL)
+					if err == nil {
+						logger.Info("Successfully set up RBAC", log.Ctx{"url": rbacAPIURL})
+						break
+					}
+
+					logger.Warn("Unable to set up RBAC, trying again in a minute", log.Ctx{"url": rbacAPIURL, "err": err})
+					time.Sleep(time.Minute)
```
Comment (stgraber) [inline lxd/daemon.go]: Current logic is a very bad idea as this means that we'd have candid setup but no RBAC, effectively allowing users to do whatever they want until rbac is up.

File: lxd/rbac/server.go [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+		},
+	}
+
+	err := agent.SetUpAuth(r.client, &authInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	r.client.Client.Jar, err = cookiejar.New(&cookiejar.Options{
+		Filename: cookiejar.DefaultCookieFile(),
```
Comment (monstermunchkin) [inline lxd/rbac/server.go]: We might also want to change this. `DefaultCookieFile()` points to `filepath.Join(homeDir(), ".go-cookies")`.

**Comment by stgraber** [comment]
Looks good to go, no rbac related errors, not that there is any test directly interacting with RBAC due to it being an external service which we can't reach or really simulate from within the testsuite.

**Comment by stgraber** [comment]
@brauner can you review?

**Pull request merged**